eng/pipelines/templates/steps/initialize-test-environment.yml

```diff
@@ -43,7 +43,7 @@ steps:
     displayName: 'Update versions for source build'
     inputs:
       scriptPath: 'eng/versioning/update_versions.py'
-      arguments: '--update-type library --build-type client --sr'
+      arguments: '--update-type library --build-type client --sr --vo $(VersionOverrides)'
```

Comment:
Let's not have this piggyback on the `TestFromSource` indicator as that controls a lot of functionality other than version updating. Let's make another switch here called something like `TestVersionSupport` where it's only ran during the special version validation runs.